From 7bc4876fae850062b06bf4e3c5a56310addf52d3 Mon Sep 17 00:00:00 2001
From: Larry Dewey <ldewey@suse.com>
Date: Tue, 7 Jan 2020 02:48:11 -0700
Subject: [PATCH] virt: adding kernel boot parameters to libvirt xml
 #55245 (#197)

* virt: adding kernel boot parameters to libvirt xml

SUSE's autoyast and Red Hat's kickstart take advantage of kernel paths,
initrd paths, and kernel boot command line parameters. These changes
provide the option of using these, and will allow salt and
autoyast/kickstart to work together.

Signed-off-by: Larry Dewey <ldewey@suse.com>

* virt: Download linux and initrd

Signed-off-by: Larry Dewey <ldewey@suse.com>
---
 salt/states/virt.py | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/salt/states/virt.py b/salt/states/virt.py
index 486aeb21b3..200c79d35c 100644
--- a/salt/states/virt.py
+++ b/salt/states/virt.py
@@ -617,13 +617,22 @@ def running(
 
         .. versionadded:: 3000
 
-    :param boot_dev:
-        Space separated list of devices to boot from sorted by decreasing priority.
-        Values can be ``hd``, ``fd``, ``cdrom`` or ``network``.
+    :param boot:
+        Specifies kernel for the virtual machine, as well as boot parameters
+        for the virtual machine. This is an optionl parameter, and all of the
+        keys are optional within the dictionary. If a remote path is provided
+        to kernel or initrd, salt will handle the downloading of the specified
+        remote fild, and will modify the XML accordingly.
 
-        By default, the value will ``"hd"``.
+        .. code-block:: python
 
-        .. versionadded:: 3002
+            {
+                'kernel': '/root/f8-i386-vmlinuz',
+                'initrd': '/root/f8-i386-initrd',
+                'cmdline': 'console=ttyS0 ks=http://example.com/f8-i386/os/'
+            }
+
+        .. versionadded:: 3000
 
     .. rubric:: Example States
 
-- 
2.29.2


